Method exclude

nemeth-z opened the pull request:
Feature implementation for #44

nemeth-z committed:
Implemented method instrumentation exclusion via environment or compiler argument

Added and extra argument to handle method exclusion from the environment (or comp arg),
extended the javac7 instrumenter with the new exclusion logic,
refactored the method translator class by moving the method exclusion logic to the instrumenter (exclusion based on the content of a method is left in the translator, like abstract method handling),
refactored the javac8 instrumenter to set the translator instead of overwriting to whole method handling logic,
introduced the Grade TestSet plugin to easily be able to define new test groups to modify compiler arguments for a given test group,
added three test class for the new logic (and to show an example on the test set usage).

Fixes #44

nemeth-z committed:
Class exclusion refactoring

Aligned the class exclusion code to the method exclusion (variable and method names should be more self explanatory now),
Added separate subsections for class and method exclusions and updated the anchors to point to the correct sub-sections.

snorbi07 commented:
Hi @nemeth-z,
Thanks for the PR, looks great! Since this changes the default behavior, could you please add a short disclaimer to the "Can you give me a more technical description?" section.

nemeth-z committed:
Documentation alignment (method exclude)

With the new exclusion mechanism in place, the default behavior has changed. Updated the technical description to make this clear and avoid possibly confusion.
